x-pack/plugins/observability_solution/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -220,7 +227,7 @@ const mapToSources = (sources: DocumentTypeData[], hasAnySourceDocBefore: boolea
       documentType,
       rollupInterval,
       hasDocs,
-      hasDurationSummaryField: allHaveDurationSummary,
+      hasDurationSummaryField: hasDocs && allHaveDurationSummary,
```

Comment:
Why is `allHaveDurationSummary` false for TransactionMetric types and incorrectly `true` for ServiceTransactionMetric types to begin with when only using < 8.7 data?

Reply:
`ServiceTransactionMetric` docs always contain `transaction.duration.summary` field. 

The [logic](https://github.com/elastic/kibana/pull/180903/files#diff-8954ac800d668cee3ec136488e3034ff94b6e7f4b42931c1767d1fde19eda359R153) assumes `allHaveDurationSummary` to be true for `ServiceTransactionMetric` when building the docType array. That's why I added this extra `hasDocs` check, to make sure the endpoint responds consistently.